sui-execution/cut/src/plan.rs

```diff
@@ -0,0 +1,626 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::env;
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use toml::value::Value;
+
+use crate::args::Args;
+
+/// Description of where packages should be copied to, what their new names should be, and whether
+/// they should be added to the [workspace] `members` or `exclude` fields.
+#[derive(Debug)]
+pub(crate) struct CutPlan(BTreeMap<String, CutPackage>);
+
+/// Details for an individual copied package in the feature being cut.
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) struct CutPackage {
+    dst_name: String,
+    src_path: PathBuf,
+    dst_path: PathBuf,
+    ws_state: WorkspaceState,
+}
+
+/// Whether the package in question is an explicit member of the workspace, an explicit exclude of
+/// the workspace, or neither (in which case it could still transitively be one or the other).
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum WorkspaceState {
+    Member,
+    Exclude,
+    Unknown,
```

Comment:
nit: not sure if later down the line we can pin down more precisely what `Unknown` represents, this is a bit more nebulous than the clarifying comment suggests :-). If it stays a transitive inclusion, we could name it `Transitive`. Or maybe `Any` to represent it could be any of the other variants? Either way, don't think this is important right now.

Reply:
Unknown means that it wasn't explicitly listed in the `members` or `exclude` lists.  This can mean it is (transitively) either a member of exclude, but it also could mean it's neither, so it really is the absence of information.  For our purposes (copying crates) this is fine though -- we only care whether something is explicitly a member or an exclude, to add its copy into the right list in the workspace toml.